### PR TITLE
Add backup command

### PR DESCRIPTION
**Pull request by moonso**
This PR adds a a backup command so that we can backup the vital parts of scout beginning as soon as this is merged. The command is `scout export database`.

**How to test**:
1. install this branch
1. run the backup command

**Expected outcome**:
A directory with zipped bson files are created. These can be used to recreate the manually inserted content.

**Review:**
- [x] code approved by dNil
- [x] tests executed by dNil

### CODE REVIEW

**Comment by moonso** [comment]
Hi, I suggest that we merge this PR asap and create a release before merging the other big PRs. In that case we can start using this backup command directly

**Comment by northwestwitch** [comment]
> Hi, I suggest that we merge this PR asap and create a release before merging the other big PRs. In that case we can start using this backup command directly

Why don't we include all the complete and ready to review PRs in the next release?

**Comment by dnil** [comment]
Sounds good to me! I'm trying to get through to testing #1548; do you have time to review @northwestwitch? Not strictly necessary, but how about a "restore" counterpart - or at least quick doc entry for loading only those tables, and a rough idea for how to reload variants, genes etc?

**Comment by northwestwitch** [comment]
Same here; ensembl biomart is down and the relative tests fails. I've fixed it here: #1632

**Comment by northwestwitch** [comment]
> Sounds good to me! I'm trying to get through to testing #1548; do you have time to review @northwestwitch? Not strictly necessary, but how about a "restore" counterpart - or at least quick doc entry for loading only those tables, and a rough idea for how to reload variants, genes etc?

**Comment by northwestwitch** [comment]
Ohi sorry, wrong button! :D

File: docs/admin-guide/backup.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        variant_collection
+        clinvar_collection
+        exon_collection
+        transcript_collection
+        filter_collection
+```
+
+IF a user want to backup the whole database use the flag `--all-collections`, however in that case we recommend to follow mongodbs [recommendations][mongodbbackup] on database backups
+
+## Recreate database
```
Comment (dnil) [inline docs/admin-guide/backup.md]: :+1: 😆

**Comment by dnil** [comment]
Eh, I'm already on it. Sorry..

**Comment by northwestwitch** [comment]
> Eh, I'm already on it. Sorry..

It's Ok

**Comment by dnil** [comment]
Relies on mongodump, it appears, so needs a requirement (or install) for that!

**Comment by moonso** [comment]
@dnil mongodump/restore are part of the MongoDB distribution

**Comment by dnil** [comment]
Depends on the platform I guess, but I suppose you are right for a server. How about adding a doc line "Ensure that the mongodump binary from your mongo distribution is in your current $PATH and executable." then?

Works like a charm otherwise!

File: docs/admin-guide/backup.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Backing up Scout
+In general it is best practice to follow the mongodb official [recommendations][mongodbbackup] for backing up a database.
+For a "light" backup there is a command in scout which works as a wrapper around [`mongodump`][mongodump]. This command was added to ease the process of backing up without including static information. Some of the collections are better to recreate than to backup including genes, transcripts and especially the gigantic variants collection.
```
Comment (dnil) [inline docs/admin-guide/backup.md]:
```suggestion
For a "light" backup there is a command in scout which works as a wrapper around [`mongodump`][mongodump]. This command was added to ease the process of backing up without including static information. Some of the collections are better to recreate than to backup including genes, transcripts and especially the gigantic variants collection. Please make sure [`mongodump`] is on your `PATH` before running this backup command.
```

File: docs/admin-guide/backup.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+## Recreate database
+
+To recreate do the following steps:
+
+1. Use a fully operational instance of scout, either an existing one or create one with `scout setup database`
+2. Run command `mongorestore --gzip path/to/dump`. Read docs for [mongorestore][mongorestore] for more information
+3. Everything should now be in place except for the variants. Add the variants for a case with `scout upload variants <case_id>`. See `--help` for more info.
+
+
+[mongorestore]: https://docs.mongodb.com/manual/reference/program/mongodump/
```
Comment (dnil) [inline docs/admin-guide/backup.md]:
```suggestion
[mongodump]: https://docs.mongodb.com/manual/reference/program/mongodump/
```

**Comment by dnil** [comment]
This is also ready to merge, pending tests..

**Pull request merged**